Branch-TestBed/Branch-SDK-Tests/BranchLoggerTests.m

```diff
@@ -11,99 +11,251 @@
-
+    __block int count = 0;
     logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
-        if ([message isEqualToString:@"[BranchSDK][Debug][BranchLoggerTests testLogLevelThresholdBlocksLowerLevels] This message should trigger the log callback."] && logLevel >= BranchLogLevelDebug) {
-            [expectation fulfill];
-        } else if (logLevel == BranchLogLevelVerbose) {
-            XCTFail();
-        }
+        count++;
     };
+    [logger logError:@"msg" error:nil];
     
-    [logger logVerbose:@"This verbose message should not trigger the log callback."];
-    [logger logDebug:@"This message should trigger the log callback."];
+    XCTAssertTrue(count == 0);
+}
 
-    [self waitForExpectationsWithTimeout:1 handler:nil];
+- (void)testLoggingEnabled_Yes {
+    BranchLogger *logger = [BranchLogger new];
+    logger.loggingEnabled = YES;
+    
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count++;
+    };
+    
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 2);
 }
 
-- (void)testLogCallbackExecutesWithCorrectParameters {
-    XCTestExpectation *expectation = [self expectationWithDescription:@"Log callback expectation"];
-    NSString *expectedMessage = @"[BranchSDK][Info][BranchLoggerTests testLogCallbackExecutesWithCorrectParameters] Test message";
-    BranchLogLevel expectedLevel = BranchLogLevelInfo;
+- (void)testLoggingIgnoresNil {
+    BranchLogger *logger = [BranchLogger new];
+    logger.loggingEnabled = YES;
+    
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count++;
+    };
+    
+    [logger logError:nil error:nil];
+    XCTAssertTrue(count == 0);
+}
 
+- (void)testLoggingIgnoresEmptyString {
     BranchLogger *logger = [BranchLogger new];
+    logger.loggingEnabled = YES;
+    
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count++;
+    };
+    
+    [logger logError:@"" error:nil];
+    XCTAssertTrue(count == 0);
+}
 
+- (void)testLoggingEnabled_YesThenNo {
+    BranchLogger *logger = [BranchLogger new];
+    logger.loggingEnabled = YES;
+    
+    __block int count = 0;
     logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
-        XCTAssertEqualObjects(message, expectedMessage, "Logged message does not match expected message.");
-        XCTAssertEqual(logLevel, expectedLevel, "Logged level does not match expected level.");
-        XCTAssertNil(error, "Error should be nil.");
-        [expectation fulfill];
+        count = count + 1;
```

Comment:
Some callbacks use `count++;`, others `count = count +1;` is there a preferred style?

Reply:
Not really. When equivalent, `count++` is most commonly used in C. Google's style guide suggests `++count` as being more readable when prefix/postfix doesn't matter. I personally prefer being explicit in tests.